thesis-de/content-de.tex

```diff
@@ -360,6 +362,128 @@ \subsubsection{Durchführung der Evaluierung}
+System darauf geachtet werden, dass die Dokumentation von Softwarekomponenten
+für den Entwickler so einfach wie möglich erfolgt.
+
+Wenn die Dokumentation einer API vollständig und das ausführende System
+den Performance-Anforderungen gerecht ist, bringt die Nutzung von einem
+API-System gewisse Vorteile. Darunter ist ein besseres Management von
+Zugriffsrechten und somit auch gleichzeitig eine Verbesserung der
+Systemsicherheit. Durch den vorab festgelegten Funktionsumfang der
+API kann der Systeminhaber geziehlt kontrollieren, welche Inhalte ein Benutzer
+sehen oder modifizieren kann. Dabei können grobe Einschränkungen, wie das Lesen
+oder Schreiben auf der gesamten Datenbank, oder auch detaillierte Einschränkung,
+wie das Lesen oder Schreiben von einer einzelnen Tabelle in der Datenbank,
+kontrolliert werden. Anders wie bei der Verbindung über einen Datenbanktreiber
+kann bei der API der potentielle Angreifer keine eigenen Datenbankbefehle
+definieren und ausführen. Somit wird das Risiko der Datenmanipulation durch eine
+zusätzliche Schicht(Layer) veringert.
+\nl%
+
+\subsubsection{Vorbereitung der Evaluierung}
+\label{subsubsec:api_vorbereitung_der_evaluierung}
+Anders wie bei der Evalierung der Datenbanksysteme wurde bei der Evaluierung
+der möglichen API-Systeme keine Liste im vorhin erstellt. Dies lag daran, dass
+keiner der Projektmitglieder oder Interessenten an dem Projekt Erfahrungen mit
+API-Systemen im vorhinein besitzen.
+
+Aufgrund dessen wurde entschieden, eine Liste von Anforderungen an das
+API-System zu erstellen und im Anschluss während der Evaluierungsphase
+Softwarelösungen zu suchen, welche die Anforderungen am besten erfüllen kann.
+
+Die festgehaltenen Anforderungen sind nun nach Relevanz absteigend aufgelistet:
+\begin{outline}
+  \1 Die API-Software muss auf einer selbst administrierten Hardware-Umgebung
+  betrieben werden können. Das verwenden von einem fertigen API-Softwarepaket
+  von einem Cloudanbietern, bei welchem der Administrator alle Konfigurationen
+  über ein Web-Interface des Providers vornimmt und der Cloudanbieter die
+  vollständige Hardware-Umgebung administriert, ist ein absolutes
+  Ausschlusskriterium. Es ist vorgehsehen, dass das API System möglichst nah 
+  an dem Datenhaltungssystem liegt, um mögliche Latenzen zwischen den beiden
+  Systemen so niedrig wie möglich zu halten. Zudem sollen externe Cloudanbieter
+  keinen Zugriff auf die teilweise Kundenbezogenen Daten der Datenhaltung
+  erhalten. Zudem kann während dem Projekt entschiedenen werden, dass das 
+  Datenhaltungssystem nicht von externen Systemen erreichbar sein soll, sodass
+  die damit verbundene Lösung über einen Cloudprovider nicht möglich wäre. 
+  Externe Systeme sind in diesem Projekt Geräte, welche außerhalb des 
+  Unternehmens betrieben werden.
+  \1 Die API-Software muss vollständig konfigurierbar und administrierbar sein.
+  Entsprechend soll die Software nur eine leere Hülle mit Standartfunktionen
```

Comment:
fixed in https://github.com/DCC-org/documents/pull/68/commits/163c9d035ece91d491ebed273b49ae7bb2028366